File: 2364-count_number_of_bad_pairs/go/main.go
package main

import (
	"fmt"
)

func countBadPairs(nums []int) int64 {
	differenceCounter := make(map[int64]int64)
	var badPairsCount int64
	for index, number := range nums {
		diff := int64(index - number)
		badPairsCount += int64(index) - differenceCounter[diff]
		differenceCounter[diff]++
	}
	return badPairsCount
}

func main() {
	nums := []int{4, 1, 3, 3}
	result := countBadPairs(nums)
	fmt.Printf("Result: %#v\n", result)
	if result == 5 {
		fmt.Println("Passed")
	} else {
		fmt.Println("Failed")
	}
}

File: 2364-count_number_of_bad_pairs/ruby/main.rb
# @param {Integer[]} nums
# @return {Integer}
def count_bad_pairs(nums)
  difference_counter = Hash.new(0)
  bad_pairs_count = 0
  nums.each_with_index do |number, index|
    diff = index - number
    bad_pairs_count += index - difference_counter[diff]
    difference_counter[diff] += 1
  end
  bad_pairs_count
end

if __FILE__ == $0
  nums = [4,1,3,3]
  result = count_bad_pairs(nums)
  puts "Result: #{result}"

  if result == 5
    puts "Passed"
  else
    puts "Failed"
  end
end

